### PR TITLE
feat(mob-ui-react-native/Components): Menu Drawer component

### PR DESCRIPTION
**Pull request by cameronolivier**
Initial Drawer component and supporting libraries and code/utils.

### CODE REVIEW

File: packages/mob-ui-react-native/docs/components/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Components Roadmap
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/docs/components/roadmap.md:1]: please move to ROADMAP.md in root

File: packages/mob-ui-react-native/src/common/components/Drawer/Drawer.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import FloatingButton from '../FloatingButton'
+import * as XPropTypes from '../../utils/propTypes.utils'
+import DrawerInner from './DrawerInner'
+
+const Drawer = ({
+  drawerContent,
+  drawerWidth,
+  drawerBackgroundColor,
+  children,
+  ...props
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/src/common/components/Drawer/Drawer.js:19]: please add trailing comma or install prettier and run :)

File: packages/mob-ui-react-native/src/common/components/Drawer/Drawer.utils.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import * as R from 'ramda'
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/src/common/components/Drawer/Drawer.utils.js:1]: please import functions individually

File: packages/mob-ui-react-native/src/common/components/FloatingButton/FloatingButton.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import { RectButton } from "react-native-gesture-handler"
+
+import styles from './FloatingButton.styles'
+
+const FloatingButton = ({
+  onPress,
+  position,
+  style,
+  children,
+  ...props
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/src/common/components/FloatingButton/FloatingButton.js:11]: please add trailing comma or run prettier

File: packages/mob-ui-react-native/src/modules/App/App.utils.test.js [hunk 5]
```diff
@@ -1,9 +1,9 @@
-import * as SUT from './App.utils'
+  import * as SUT from './App.utils'
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/src/modules/App/App.utils.test.js:1]: please remove spaces

File: packages/mob-ui-react-native/src/modules/DrawerExample/DrawerExample.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { Text, View } from 'react-native'
+import { enableScreens } from 'react-native-screens'
+import styles from './DrawerExample.styles'
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/src/modules/DrawerExample/DrawerExample.js:4]: please add newline between npm and local deps

File: packages/mob-ui-react-native/src/modules/DrawerExample/DrawerExample.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+enableScreens()
+
+const DrawerExample = () => {
+  return (
+    <Drawer
+      drawerWidth='100%'
+      drawerBackgroundColor='#fecea8'
+      drawerContent={<>
+        <Text style={{fontStyle: 'italic', fontSize: 24, fontWeight: 'bold' }}>Peek-a-boo!</Text>
```
Comment (neilrussell6) [inline packages/mob-ui-react-native/src/modules/DrawerExample/DrawerExample.js:15]: ideally we avoid any forms of personal expression and personality in the sample text we use :)
that is also descriptive eg.
eg.
```
<Text>DRAWER CONTENT</Text>
<Text>DRAWER CONTENT CHILDREN</Text>
```

**Review by neilrussell6 — CHANGES_REQUESTED** [review]

File: packages/mob-ui-react-native/src/common/components/Drawer/Drawer.utils.js (hunk 3 above)
Comment (cameronolivier) [reply packages/mob-ui-react-native/src/common/components/Drawer/Drawer.utils.js:1]: whhyyyyyy are we doing this? it's such a pain to use this way :/ (this is an offline discussion, I suppose... )

File: packages/mob-ui-react-native/src/modules/DrawerExample/DrawerExample.js (hunk 7 above)
Comment (cameronolivier) [reply packages/mob-ui-react-native/src/modules/DrawerExample/DrawerExample.js:15]: but it's cute. 


....ok ok I'll change it. Was more for the video Demo :D

File: packages/mob-ui-react-native/src/common/components/FloatingButton/FloatingButton.js (hunk 4 above)
Comment (cameronolivier) [reply packages/mob-ui-react-native/src/common/components/FloatingButton/FloatingButton.js:11]: there's no trailing comma with spreads :) (I have prettier running)

**Pull request closed without merging**